core/src/Kokkos_UniqueToken.hpp

```diff
@@ -126,7 +126,7 @@ class AcquireTeamUniqueToken {
   using size_type        = typename token_type::size_type;
   using team_member_type = typename TeamPolicy::member_type;
   using scratch_view =
-      Kokkos::View<size_type, typename exec_space::scratch_memory_space,
+      Kokkos::View<size_type, typename exec_space::scratch_memory_space_l0,
```

Comment:
Would we want to give users the option to put the Token in L1 and use the L0 for something else?

Reply:
That would be a different discussion. 🙂 This pull request doesn't change where scratch memory is allocated